src/Illuminate/Support/Arr.php

```diff
@@ -456,4 +456,69 @@ public static function where($array, callable $callback)
 
         return $filtered;
     }
+
+    /**
+     * Index the array by array of keys.
+     *
+     * @param  array  $data
+     * @param  array  $keys
+     * @param  bool|true  $unique
+     * @return array
+     */
+    public static function getIndexedByKeys(array $data, array $keys, $unique = true)
+    {
+        $result = [];
+
+        foreach ($data as $value) {
+            static::indexByKeys($result, $value, $keys, $unique);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param  array  $result
+     * @param  array  $toSave
+     * @param  array  $keys
+     * @param  bool|true $unique
+     */
+    protected static function indexByKeys(array &$result, array $toSave, array $keys, $unique = true)
+    {
+        foreach ($keys as $key) {
+            if (! isset($result[$toSave[$key]])) {
+                $result[$toSave[$key]] = [];
+            }
+            $result = &$result[$toSave[$key]];
+        }
+
+        if ($unique) {
+            $result = $toSave;
+        } else {
+            $result[] = $toSave;
+        }
+    }
+
+    /**
+     * Converts array of arrays to one-dimensional array, where key is $keyName and value is $valueName.
+     *
+     * @param  array  $array
+     * @param  string  $keyName
+     * @param  string|array  $valueName
+     * @return array
+     */
+    public static function getIndexedValues(array $array, $keyName, $valueName)
+    {
+        $result = [];
+        foreach ($array as $value) {
+            if (is_array($valueName)) {
```

Comment:
Loop-invariant code - you should better move this condition out of `foreach` and make two different cycles